### PR TITLE
Log request ID and extended request ID with the requestId logger and …

### PR DESCRIPTION
**Pull request by zoewangg**
…request logger

## Motivation and Context
#2876 

## Description
Log request ID and extended request ID with the requestId logger and

## Testing
Tested locally.
```
2021-12-16 15:50:25,457 [main] DEBUG software.amazon.awssdk.request - Received failed response: 400, Request ID: xxxxx, Extended Request ID: not available

2021-12-16 15:50:46,493 [main] DEBUG software.amazon.awssdk.requestId - Received successful response: 200, Request ID: xxxx, Extended Request ID: not available
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/AwsXmlPredicatedResponseHandler.java [hunk 1]
```diff
@@ -94,6 +95,7 @@ public AwsXmlPredicatedResponseHandler(
 
         AwsXmlUnmarshallingContext parsedResponse = parseResponse(httpResponse, executionAttributes);
         parsedResponse = decorateContextWithError.apply(parsedResponse);
+        logRequestId(httpResponse);
```
Comment (millems) [inline core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/AwsXmlPredicatedResponseHandler.java:98]: Is this not handled by the combined response handler below?

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/CombinedResponseHandler.java [hunk 2]
```diff
@@ -68,6 +69,8 @@ public CombinedResponseHandler(HttpResponseHandler<OutputT> successResponseHandl
                                              ExecutionAttributes executionAttributes)
             throws IOException, InterruptedException {
 
+        logRequestId(httpResponse);
```
Comment (millems) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/CombinedResponseHandler.java:72]: Any specific reason to have picked this response handler to do the logging?

File: core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/AwsXmlPredicatedResponseHandler.java (hunk 1 above)
Comment (zoewangg) [reply core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/AwsXmlPredicatedResponseHandler.java:98]: Yeah, it seems AwsXmlPredicatedResponseHandler is used by s3, and combined response handler is used by all other services. (I just learned it yesterday)

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/CombinedResponseHandler.java (hunk 2 above)
Comment (zoewangg) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/CombinedResponseHandler.java:72]: This is where we log the response status code currently, so I just included the request ID here

**Review by millems — APPROVED** [review]

**Pull request merged**